src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,84 @@
+package duke.ui;
+
+import java.util.Scanner;
+
+/**
+ * The class that handles ui functions such as
+ * printing messages and errors to user and reading from user.
+ */
+public class Ui {
+    private Scanner scanner;
+
+    /**
+     * Constructor of Ui class.
+     * Instantiates an Ui object that by default reads input from System.in
+     */
+    public Ui() {
+        this.scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Shows welcome message of Duke.
+     */
+    public void showWelcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+
+        System.out.println("Hello from\n" + logo);
+    }
+
+    /**
+     * Reads user inputs.
+     * @return String of the line read from standard in.
+     */
+    public String readLine() {
+        String input = scanner.nextLine();
+        return input;
+    }
+
+    /**
+     * Greets user.
+     * @param name User name to display.
+     */
+    public void greeting(String name) {
```

Comment:
Perhaps you can consider naming your method with a verb? Maybe something like `greetUser` or `sayGreeting`.